sensing/autoware_pointcloud_preprocessor/src/concatenate_data/concatenate_and_time_sync_node.cpp

```diff
@@ -0,0 +1,458 @@
+        }
+      }
+    }
+  }
+
+  diagnostic_collector_info_ = collector_info;
+
+  diagnostic_topic_to_original_stamp_map_ = concatenated_cloud_result.topic_to_original_stamp_map;
+  diagnostic_updater_.force_update();
+
+  // add processing time for debug
+  if (debug_publisher_) {
+    const double cyclic_time_ms = stop_watch_ptr_->toc("cyclic_time", true);
+    const double processing_time_ms = stop_watch_ptr_->toc("processing_time", true);
+    debug_publisher_->publish<tier4_debug_msgs::msg::Float64Stamped>(
+      "debug/cyclic_time_ms", cyclic_time_ms);
+    debug_publisher_->publish<tier4_debug_msgs::msg::Float64Stamped>(
+      "debug/processing_time_ms", processing_time_ms);
+
+    for (const auto & [topic, stamp] : concatenated_cloud_result.topic_to_original_stamp_map) {
+      const auto pipeline_latency_ms = (this->get_clock()->now().seconds() - stamp) * 1000;
+      debug_publisher_->publish<tier4_debug_msgs::msg::Float64Stamped>(
+        "debug" + topic + "/pipeline_latency_ms", pipeline_latency_ms);
+    }
+  }
+}
+
+void PointCloudConcatenateDataSynchronizerComponent::manage_collector_list()
+{
+  std::lock_guard<std::mutex> cloud_collectors_lock(cloud_collectors_mutex_);
+
+  for (auto it = cloud_collectors_.begin(); it != cloud_collectors_.end();) {
+    if ((*it)->concatenate_finished()) {
+      it = cloud_collectors_.erase(it);  // Erase and move the iterator to the next element
+    } else {
+      ++it;  // Move to the next element
+    }
+  }
+}
+
+std::string PointCloudConcatenateDataSynchronizerComponent::format_timestamp(double timestamp)
+{
+  std::ostringstream oss;
+  oss << std::fixed << std::setprecision(9) << timestamp;
+  return oss.str();
+}
+
+void PointCloudConcatenateDataSynchronizerComponent::check_concat_status(
+  diagnostic_updater::DiagnosticStatusWrapper & stat)
+{
+  if (publish_pointcloud_ || drop_previous_but_late_pointcloud_) {
+    stat.add(
+      "concatenated cloud timestamp", format_timestamp(current_concatenate_cloud_timestamp_));
+
+    if (diagnostic_collector_info_.strategy_type == CollectorStrategyType::Naive) {
+      stat.add(
+        "first cloud's arrival timestamp", format_timestamp(diagnostic_collector_info_.timestamp));
+    } else if (diagnostic_collector_info_.strategy_type == CollectorStrategyType::Advanced) {
+      stat.add(
+        "reference timestamp min",
+        format_timestamp(
+          diagnostic_collector_info_.timestamp - diagnostic_collector_info_.noise_window));
+      stat.add(
+        "reference timestamp max",
+        format_timestamp(
+          diagnostic_collector_info_.timestamp + diagnostic_collector_info_.noise_window));
+    }
+
+    bool topic_miss = false;
+
+    int concatenated_cloud_status = 1;  // Status of concatenated cloud, 1: success, 0: failure
+    for (const auto & topic : params_.input_topics) {
+      int cloud_status = 1;  // Status of each lidar's pointcloud
```

Comment:
Same as above

Reply:
Thanks, fixed in https://github.com/autowarefoundation/autoware.universe/pull/8300/commits/f278aea13ae2a98a784429511604d3534b163fc1